### PR TITLE
Bd Creating and Controller Update

### PR DESCRIPTION
**Pull request by Stupid-Fox**

### CODE REVIEW

File: CoffeeHouse.Api/Controllers/CoffeeController.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        [HttpPost]
+        public async Task<ActionResult<IEnumerable<Coffee>>> Post(Coffee coffee)
+        {
+            _context.Coffees.Add(coffee);
+            await _context.SaveChangesAsync();
+            return Ok(coffee);
+        }
+
+        [HttpPut]
+        public async Task<ActionResult<IEnumerable<Coffee>>> Put(Coffee request)
```
Comment (stas1408123) [inline CoffeeHouse.Api/Controllers/CoffeeController.cs]: Check method update

**Review by stas1408123 — CHANGES_REQUESTED** [review]

File: CoffeeHouse.Api/Controllers/CoffeeController.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    {
+        private readonly CoffeeContext _context;
+
+        public CoffeeController(CoffeeContext context)
+        {
+            _context = context;
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Coffee>>> GetAll()
```
Comment (hromau) [inline CoffeeHouse.Api/Controllers/CoffeeController.cs]: delete all async, make ur code sync. get rid action result

File: CoffeeHouse.Api/Controllers/CoffeeController.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        {
+            return Ok(await _context.Coffees.ToListAsync());
```
Comment (hromau) [inline CoffeeHouse.Api/Controllers/CoffeeController.cs]: u must have to return object or objects, not Ok

File: CoffeeHouse.Api/Models/Coffee.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    public class Coffee
+    {
+        public int Id { get; set; }
+        public string Name { get; set; } = "Coffee";
+        public int Cost { get; set; }
+        public int Temperature { get; set; }
+        public int MakeTime { get; set; }
+
+
+
```
Comment (hromau) [inline CoffeeHouse.Api/Models/Coffee.cs]: ?

File: CoffeeHouse.Api/Models/CoffeeContext.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    public class CoffeeContext : DbContext
+    {
+     
+        public DbSet<Coffee> Coffees { get; set; } = null!;
+        public CoffeeContext(DbContextOptions<CoffeeContext> options)
+            : base(options)
+        {
+            Database.EnsureCreated();   
+
+
```
Comment (hromau) [inline CoffeeHouse.Api/Models/CoffeeContext.cs]: ?

File: CoffeeHouse.Api/Program.cs [hunk 6]
```diff
@@ -1,4 +1,21 @@
-var builder = WebApplication.CreateBuilder(args);
+using CoffeeHouse.Api.Models;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore;
+
+
+
+
```
Comment (hromau) [inline CoffeeHouse.Api/Program.cs]: what the spaces?

File: CoffeeHouse.Api/Program.cs [hunk 7]
```diff
@@ -1,4 +1,21 @@
...
+
+
+
+var builder = WebApplication.CreateBuilder();
+
+// �������� ������ ����������� �� ����� ������������
```
Comment (hromau) [inline CoffeeHouse.Api/Program.cs]: ïîëó÷àåì ñòðîêó ïîäêëþ÷åíèÿ èç ôàéëà êîíôèãóðàöèè

File: CoffeeHouse.Api/Program.cs [hunk 8]
```diff
@@ -1,4 +1,21 @@
...
+string connection = builder.Configuration.GetConnectionString("DefaultConnection");
+
+// ��������� �������� ApplicationContext � �������� ������� � ����������
```
Comment (hromau) [inline CoffeeHouse.Api/Program.cs]: äîáàâëÿåì êîíòåêñò ApplicationContext â êà÷åñòâå ñåðâèñà â ïðèëîæåíèå

File: CoffeeHouse.Api/Program.cs [hunk 9]
```diff
@@ -8,18 +25,21 @@
...
 // Configure the HTTP request pipeline.
 if (app.Environment.IsDevelopment())
 {
     app.UseSwagger();
     app.UseSwaggerUI();
 }
 
+
+
+// ��������� �������� ApplicationContext � �������� ������� � ����������
```
Comment (hromau) [inline CoffeeHouse.Api/Program.cs]: äîáàâëÿåì êîíòåêñò ApplicationContext â êà÷åñòâå ñåðâèñà â ïðèëîæåíèå

File: CoffeeHouse.Api/appsettings.json [hunk 10]
```diff
@@ -1,9 +1,13 @@
...
-    "LogLevel": {
-      "Default": "Information",
-      "Microsoft.AspNetCore": "Warning"
-    }
-  },
-  "AllowedHosts": "*"
-}
+
+  "ConnectionStrings": {
+    "DefaultConnection": "Server=(localdb)\\mssqllocaldb;Database=mobilesdb;Trusted_Connection=True;",
```
Comment (hromau) [inline CoffeeHouse.Api/appsettings.json]: change local db to mssql

File: CoffeeHouse.Api/Program.cs [hunk 11]
```diff
@@ -1,4 +1,21 @@
-var builder = WebApplication.CreateBuilder(args);
+using CoffeeHouse.Api.Models;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore;
```
Comment (hromau) [inline CoffeeHouse.Api/Program.cs]: second time is extra

**Review by hromau — CHANGES_REQUESTED** [review]

File: CoffeeHouse.Api/Controllers/CoffeeController.cs [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+        }
+
+        [HttpGet]
+        public IEnumerable<Coffee> GetAll()
+        {
+            return _context.Coffees.ToList();
+        }
+
+        [HttpGet("{id}")]
+        public Coffee Get(int id)
```
Comment (hromau) [inline CoffeeHouse.Api/Controllers/CoffeeController.cs]: change

**Review by hromau — CHANGES_REQUESTED** [review]

File: CoffeeHouse.Api/Controllers/CoffeeController.cs [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+
+        [HttpPost]
+        public void Post(Coffee coffee)
+        {
+            _context.Coffees.Add(coffee);
+            _context.SaveChanges();     
+        }
+
+        [HttpPut]
+        public void Put(Coffee request)
```
Comment (stas1408123) [inline CoffeeHouse.Api/Controllers/CoffeeController.cs]: Rename parametr

File: CoffeeHouse.Api/Controllers/CoffeeController.cs [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+        {
+            request.Cost = 10;
```
Comment (stas1408123) [inline CoffeeHouse.Api/Controllers/CoffeeController.cs]: Delete hard code 
You can use swagger for testing

**Review by stas1408123 — CHANGES_REQUESTED** [review]

**Review by stas1408123 — APPROVED** [review]

File: CoffeeHouse.Api/Controllers/CoffeeController.cs [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+        public Coffee GetCoffee(int id)
+        {
+            var coffee = _context.Coffees.Find(id);
+            if (coffee == null)
+                throw new ArgumentNullException("Object cannot be null", nameof(coffee));
+            return coffee;
+        }
+
+        [HttpPost]
+        public void PostCoffee(Coffee coffee)
```
Comment (hromau) [inline CoffeeHouse.Api/Controllers/CoffeeController.cs]: rename

**Review by hromau — CHANGES_REQUESTED** [review]

File: CoffeeHouse.Api/Controllers/CoffeeController.cs [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+
+        [HttpPost]
+        public void AddNewCoffee(Coffee coffee)
+        {
+            _context.Coffees.Add(coffee);
+            _context.SaveChanges();     
+        }
+
+        [HttpPut]
+        public void PutCoffee(Coffee coffee)
```
Comment (hromau) [inline CoffeeHouse.Api/Controllers/CoffeeController.cs]: it shouldn't be PutCofee. let's think about something else

**Review by hromau — CHANGES_REQUESTED** [review]

File: CoffeeHouse.Api/Controllers/CoffeeController.cs (hunk 16 above)
Comment (hromau) [reply CoffeeHouse.Api/Controllers/CoffeeController.cs]: rename all methods. make all is understandable

File: CoffeeHouse.Api/Program.cs [hunk 17]
```diff
@@ -22,4 +25,6 @@
 
 app.MapControllers();
 
+app.MapGet("/", (CoffeeContext db) => db.Coffees.ToList());
```
Comment (hromau) [inline CoffeeHouse.Api/Program.cs:28]: what is the strange routing?

**Review by hromau — APPROVED** [review]

**Pull request merged**